test/userController.test.js

```diff
@@ -58,9 +67,35 @@ afterAll(async () => {
   await mongodb.stop();
 });
 
+beforeEach(async () => {
+  // Delete all users before creating the test user
+  await User.deleteMany({});
+
+  // Create a test user before each test
+  testUser = await User.create({
+    name: 'ava',
+    username: 'ava',
+    email: 'ava@ava.com',
+    password: 'secret',
+    role: 'user',
+  });
+  testAdmin = await User.create({
+    name: 'admin',
+    username: 'admin',
+    email: 'admin@admin.com',
+    password: 'secret',
+    role: 'admin',
+  });
+});
+
+afterEach(async () => {
+  // Delete the test user after each test
+  await User.deleteMany({});
+});
```

Comment:
We can delete this :+1: 

Jest will run in the following order within a test block/describe block (see: [https://noriste.github.io/reactjsday-2019-testing-course/book/jest-101/jest-lifecycle.html)](https://noriste.github.io/reactjsday-2019-testing-course/book/jest-101/jest-lifecycle.html))

1.  beforeAll
2.  beforeEach
3.  the test itself
4.  afterEach
5.  repeat 2-3 until all tests in this block have completed
6.  afterAll 

So if we are already going to delete all the users in step (2) `beforeEach` it doesn't add anything to also delete them in step (4) `afterEach` . Since we have the rest of our logic in `beforeEach` we can simplify our code here by just removing `afterEach` altogether